manim/utils/tex_file_writing.py

```diff
@@ -50,22 +50,31 @@ def tex_to_svg_file(
     """
     if tex_template is None:
         tex_template = config["tex_template"]
-    tex_file = generate_tex_file(expression, environment, tex_template)
+    tex_file, svg = generate_tex_file(expression, environment, tex_template)
```

Comment:
I am not sure whether it was necessary to change the output type of `generate_tex_file`: in the current implementation, as far as I am aware, this method always returns the path of a given `.tex` file. We can just use `tex_file.with_suffix(".svg")` to get the path to the -- if it exists -- corresponding svg file. If it exists (which we can and should check here!), `compile_tex` and `convert_to_svg` could be skipped (which is good, as the cleaned state would then be kept).